sdk/test-utils/perfstress/GettingStarted.md

```diff
@@ -0,0 +1,258 @@
+# Writing Performance Tests
+
+## [Index](#index)
+
+- [Setting up the project](#Setting-up-the-project)
+  - [Track 2](#Setting-up-the-project)
+  - [Track 1](#For-perf-testing-track-1-version-of-the-same-package)
+- [Writing perf tests](#writing-perf-tests)
+  - [Entry Point](#Entry-point)
+  - [Base Class](#Base-Class)
+  - [Test File](#Test-file)
+  - [Custom Options](#custom-options)
+- [Executing the perf tests](#executing-the-perf-tests)
+  - [Command to run](#Command-to-run)
+  - [Adding Readme/Instructions](#Adding-Readme/Instructions)
+  - [Testing an older track 2 version](#Testing-an-older-track-2-version)
+
+## [Setting up the project](#Setting-up-the-project)
+
+To add perf tests for the `sdk/<service>/<service-sdk>` package, follow the steps below.
+
+1. Create a new folder for the perf tests.
+
+   Path- `sdk/<service>/perf-tests/<service-sdk>`
+
+   (Create the `perf-tests` folder if that doesn't exist)
+
+2. This new perf test project will be managed by the rush infrastructure in the repository, with the package name `@azure-tests/<service-sdk>`. To allow rush to manage the project, add the following entry in `rush.json`
+
+   ```
+       {
+         "packageName": "@azure-tests/<service-sdk>",
+         "projectFolder": "sdk/<service>/perf-tests/<service-sdk>",
+         "versionPolicyName": "test"
+       }
+
+   ```
+
+3. Tests will live under `sdk/<service>/perf-tests/<service-sdk>/test`
+4. Add a `package.json` such as [example-perf-package.json](https://github.com/Azure/azure-sdk-for-js/blob/fe9b1e5a50946f53b6491d7f67b2420d8ee1b229/sdk/storage/perf-tests/storage-file-datalake/package.json) at `sdk/<service>/perf-tests/<service-sdk>` folder.
+
+   Make sure to import your `<service-sdk>` and the `test-utils-perfstress` project.
+
+   ```json
+     "devDependencies": {
+        "@azure/<service-sdk>": "^<version-in-master-branch>",
+        "@azure/test-utils-perfstress": "^1.0.0"
+      }
+   ```
+
+   _Note: `"@azure/test-utils-perfstress"` is not a published npm package._
+
+   Set the name of the package and mark it as private.
+
+   ```json
+    "name": "@azure-tests/perf-<service-sdk>",
+    "private": true,
+   ```
+
+5. Run `rush update` and commit the changes to the `pnpm-lock` file.
+6. Copy the `tsconfig.json`, `sample.env`(and `.env`) files that are present at the `sdk/<service>/<service-sdk>` to `sdk/<service>/perf-tests/<service-sdk>`.
```

Comment:
NIT: we do not really need to emit js files if we can run the tests using ts-node. Also note that the package's tsconfig will extend the central one in the root of the repo so this line will need to be edited for the perftest package to accommodate its relative location.

Reply:
> we do not really need to emit js files if we can run the tests using ts-node

The build step is run in the CI if the package is touched to make sure the perf tests compile, that'd generate js files.

> Also note that the package's tsconfig will extend the central one in the root of the repo so this line will need to be edited for the perftest package to accommodate its relative location.

Good point. I did think about it, but I didn't want to spoonfeed. I'll add it anyway.

Reply:
I meant to still typecheck but just add [`noEmit`: true](https://github.com/Azure/azure-sdk-for-js/blob/7aa05cc212ba3b92a1e7c4e2e9f0b53dc036c4b0/sdk/textanalytics/perf-tests/text-analytics/tsconfig.json#L8).

Reply:
Ah.. I see. Good point, I did not know `noEmit`. Will add.

Reply:
I'm not explicitly mentioning noEmit since there is no harm in generating the files.
But I have added it in the example tsconfig.json below.